src/patternfly/components/DataList/data-list.scss

```diff
@@ -123,48 +130,49 @@
 
   // Compact
   --pf-c-data-list--m-compact--FontSize: var(--pf-global--FontSize--sm);
+  --pf-c-data-list--m-compact--LineHeight: var(--pf-global--LineHeight--sm);
   --pf-c-data-list--m-compact__check--FontSize: var(--pf-global--FontSize--md);
-  --pf-c-data-list--m-compact__cell--PaddingTop: var(--pf-global--spacer--sm);
+  --pf-c-data-list--m-compact__cell--PaddingTop: calc(var(--pf-global--spacer--sm) + (var(--pf-c-data-list--LineHeight) * 1px));
   --pf-c-data-list--m-compact__cell--PaddingBottom: var(--pf-global--spacer--sm);
   --pf-c-data-list--m-compact__cell--md--PaddingBottom: 0;
   --pf-c-data-list--m-compact__cell-cell--PaddingTop: 0;
   --pf-c-data-list--m-compact__cell-cell--md--PaddingTop: var(--pf-global--spacer--sm);
   --pf-c-data-list--m-compact__cell--cell--MarginRight: var(--pf-global--spacer--md);
-  --pf-c-data-list--m-compact__item-control--PaddingTop: var(--pf-global--spacer--sm);
+  --pf-c-data-list--m-compact__item-control--PaddingTop: calc(var(--pf-global--spacer--sm) + (var(--pf-c-data-list--LineHeight) * 1px));
   --pf-c-data-list--m-compact__item-control--PaddingBottom: 0;
   --pf-c-data-list--m-compact__item-control--MarginRight: var(--pf-global--spacer--md);
-  --pf-c-data-list--m-compact__item-action--PaddingTop: var(--pf-global--spacer--sm);
-  --pf-c-data-list--m-compact__item-action--PaddingBottom: var(--pf-global--spacer--sm);
+  --pf-c-data-list--m-compact__item-action--PaddingTop: 0;
```

Comment:
This is causing this kind of layout

<img width="900" alt="Screen Shot 2020-12-14 at 6 12 48 PM" src="https://user-images.githubusercontent.com/35148959/102151216-00f3cb00-3e38-11eb-8a3e-06b3194704e7.png">